components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -123,13 +128,31 @@ public static void addCartridge(CartridgeBean cartridgeDefinition) throws RestAP
                         "Successfully added cartridge: [cartridge-type] %s ",
                         cartridgeDefinition.getType()));
             }
-        } catch (Exception e) {
+        }catch (CloudControllerServiceCartridgeAlreadyExistsExceptionException e) {
+            String msg = "Could not add cartridge";
+            log.error(msg, e);
+            throw new RestAPIException(msg);
+        } catch (CloudControllerServiceInvalidCartridgeDefinitionExceptionException e) {
```

Comment:
If there's nothing specific to be logged when each of these exceptions was caught, might be a good option to use "|" notation to have them all clustered into one catch block, which would improve readability of the code.